haskell/test/SMT/Common/Run.hs

```diff
@@ -0,0 +1,219 @@
+
+        let PropertyTestInput (contractCalls, ms) = mkPropertyTestInput $ PropertyTestArg
+              { ptaGuardianAddr = unTAddress guardianContract
+              , ptaGovAddr = unTAddress tokenContract
+              , ptaViewContractAddr = registryDaoConsumer
+              }
+
+        advanceLevel (ms & msLevel)
+
+        -- Modifying `FullStorage` used by Registry/TreasuryDao
+        -- Update `msContracts` to track the contract called used by haskell implementation
+        let newMs = ms
+              { msFullStorage = mptoModifyFs (ms & msFullStorage)
+              , msContracts = Map.fromList
+                  [ ((unTAddress tokenContract), SimpleFA2ContractType $ SimpleFA2Contract [] (toMutez 0))
+                  , ((unTAddress registryDaoConsumer), OtherContractType $ OtherContract [] (toMutez 0))
+                  ]
+              }
+
+        -- Originate Dao for nettest
+        let
+          originateData = UntypedOriginateData
+            { uodName = "BaseDAO"
+            , uodBalance = toMutez 0
+            , uodStorage = untypeValue $ toVal $ (newMs & msFullStorage)
+            , uodContract = convertContract baseDAOContractLigo
+            }
+
+        daoUntyped <- originateUntyped originateData
+        let dao = TAddress @Parameter daoUntyped
+
+        -- The registry/treasury dao can run out of mutez
+        when (mptoContractType == RegistryDaoContract || mptoContractType == TreasuryDaoContract) $ do
+          sendXtz daoUntyped (unsafeBuildEpName "callCustom") ([mt|receive_xtz|], lPackValueRaw ())
+
+        let newMs_ = newMs { msSelfAddress = daoUntyped }
+
+        callEntrypointAndCompare (dao, tokenContract, registryDaoConsumer) contractCalls newMs_
+
+    )
+    emulatedImplIntegrational
+
+callEntrypointAndCompare :: (TAddress Parameter, TAddress FA2.Parameter, TAddress (MText, (Maybe MText))) -> [ModelCall] -> ModelState -> EmulatedT PureM ()
+callEntrypointAndCompare _ [] _ = pure ()
+callEntrypointAndCompare (dao, gov, viewC) (mc:mcs) ms = do
+
+  let (haskellErrors, updatedMs) = runBaseDaoHaskell mc ms
+      haskellStore = updatedMs & msFullStorage & fsStorage
+      govContract = updatedMs & msContracts
+        & Map.lookup (unTAddress gov)
+        & fromMaybe (error "Governance contract does not exist")
+      haskellGovStore = case govContract of SimpleFA2ContractType c -> c & sfcStorage; _ -> error "Shouldn't happen."
+      haskellGovBalance = case govContract of SimpleFA2ContractType c -> c & sfcMutez; _ -> error "Shouldn't happen."
+      viewContract = updatedMs & msContracts
+        & Map.lookup (unTAddress viewC)
+        & fromMaybe (error "View contract does not exist")
+      haskellViewStore = case viewContract of OtherContractType c -> c & ocStorage; _ -> error "Shouldn't happen."
+
+  (ligoErrors, ligoFs, ligoGovStore, ligoGovBalance, ligoViewStore) <- runBaseDaoLigo (dao, gov, viewC) mc
```

Comment:
I really have to ask you to provide much more documentation, some of the function names are pretty ambiguous, for example both `runBaseDaoLigo` and `runBaseDaoHaskell` might do similar things, one has to look at their implementation to figure out what does what.

Same about some of the operations and values here. I do understand that likely `govContract` is the contract holding the governance token, however that's because I'm very familiar with the codebase.
If someone else, or us in a few months, look at this it's hardly possible to tell what's happening.

Please with this in mind go through the code in this PR and try to document it better.

Reply:
I've renamed most of the functions and type to match what they are intended to do.

Also added more comments to the functions in `SMT.Common.Run` to explains what each function does as well. Let me know if there are still things that are confusing.